models/stg_shopify__metafield.sql

```diff
@@ -38,7 +38,10 @@ final as (
         {{ dbt_date.convert_timezone(column='cast(updated_at as ' ~ dbt.type_timestamp() ~ ')', target_tz=var('shopify_timezone', "UTC"), source_tz="UTC") }} as updated_at,
         {{ dbt_date.convert_timezone(column='cast(_fivetran_synced as ' ~ dbt.type_timestamp() ~ ')', target_tz=var('shopify_timezone', "UTC"), source_tz="UTC") }} as _fivetran_synced,
         lower({{ dbt.concat(["namespace","'_'","key"]) }}) as metafield_reference,
-        row_number() over (partition by id order by updated_at desc) = 1 as is_most_recent_record,
+        case when id is null
+            then row_number() over(partition by source_relation order by updated_at desc) = 1
+            else row_number() over(partition by id, source_relation order by updated_at desc) = 1
+        end as is_most_recent_record,
```

Comment:
Same note here, we should include the order by field in the `null` case when logic.

```suggestion
        case when id is null and updated_at is null
            then row_number() over(partition by source_relation) = 1
            else row_number() over(partition by id, source_relation order by updated_at desc) = 1
        end as is_most_recent_record,
```

Reply:
updated

Reply:
An `order by` is actually required on partitions in Databricks and Snowflake, so I added an `order by source relation` to the partition.